server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -96,42 +124,68 @@ public void define(Service service) {
             dataType = "String",
             defaultValue = "none",
             example = "user@james.org",
-            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails")
+            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails"),
+        @ApiImplicitParam(
+            required = true,
+            dataType = "String",
+            name = "action",
+            paramType = "query",
+            example = "?action=restore",
+            value = "Compulsory. Needs to ve a valid action represent for an operation to perform on the Deleted Message Vault")
     })
     @ApiResponses(value = {
         @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
         @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - user param is invalid"),
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
     private TaskIdDto restore(Request request, Response response) throws JsonExtractException {
+        validateActionQuery(request, Action.RESTORE);
+
         User userToRestore = extractUser(request);
         QueryDTO queryDTO = jsonExtractor.parse(request.body());
         Query query = translate(queryDTO);
         TaskId taskId = taskManager.submit(new DeletedMessagesVaultRestoreTask(vaultRestore, userToRestore, query, queryDTO));
         return TaskIdDto.respond(response, taskId);
     }
 
+    private void validateActionQuery(Request request, Action appropriateAction) {
```

Comment:
Well to be honest the evolution here will be accepting **both** restore and export as valid action on this endpoint.

So the *appropriateActionMakes no sens*. We should ratherreturn the Action the user requested. If we succeed to parse it, we know that it will be valid.